internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,163 @@ import (
+
+	// Review: if this is just for the `init` actor then it should not be here.
+	// Review: the init actor is a special system actor, its ok for it to have a different interface with a bit more power.
+    // CreateActor(
+    //     cid                actor.StateCID
+    //     a                  addr.Address
+    //     constructorParams  actor.MethodParams
+    // )
+
+	// Review: we provide this through the Storage()
+    // IpldGet(c ipld.CID) union {Bytes, error}
+	// IpldPut(x ipld.Object) ipld.CID
+	// Review: is the expectation fo storage to be global or actor specific? this has security implications down the line
+	Storage() Storage
+}
+
+// Review: more powerfull contract for the initactor.
+type innitInvocationContext interface {
+	InvocationContext
+	// Review: the spec has no error on this? what if the addr is already taken?
+	CreateActor(cid cid.Cid, addr address.Address, constructorParams MethodParams)
+}
+
+// Storage2 defines the storage module exposed to actors.
+// Review: check the old `Storage` for some more comments on why this looks like it does
+type Storage2 interface {
+	Put(interface{}) (cid.Cid, error)
+	Get(cid.Cid, interface{}) error
+}
+
+// Epoch is the current chain epoch.
+// Dragons: this is just to match names on the VM, Height got renamed to Epoch on the spec
+// TODO: rename types.BlockHeight to block.Epoch (issue: ???)
+type Epoch types.BlockHeight
+
+// Randomness is a string of random bytes
+// Review: placeholder for matching newtypes with the spec.
+type Randomness []byte
+
+// InvocInput are the params to invoke a method in an Actor.
+// Review: why is this a struct instead of arguments?
+type InvocInput struct {
+    To      address.Address
+    Method  types.MethodID
+    Params  MethodParams
+    Value   types.AttoFIL
+}
+
+// ExitCode is the exit code of a method executing inside the VM.
+// Review: do we have reserved codes?
+// TODO: this has type UVarint in the spec
+type ExitCode types.Int64
+
+// InvocOutput is waht os returned by an actor executing a method.
+// Review: multiple returns vs a return struct. There are some minor implications with swapping vm implementations written in another language.
+type InvocOutput struct {
+	ExitCode     ExitCode
+    ReturnValue  []byte
+}
+
+// MethodParam is the parameter to an Actor method already encoded for execution in the VM.
+type MethodParam []byte
+// MethodParams is a list of `MethodParam` to be passed into an Actor method.
+type MethodParams []MethodParam
+
 // Runtime defines the ABI interface exposed to actors.
 type Runtime interface {
+	// Review: this is not in the spec anymore
 	Message() *types.UnsignedMessage
+
+	// Review: the spec splits this up and brings the get/put to be first class
 	Storage() Storage
+
+	// Review: changed from arguments to a struct
 	Send(to address.Address, method types.MethodID, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
+
+	// Review: this is not in the spec anymore
```

Comment:
This got moved to `_generateActorAddress` in `runtime.go`, but I think the public handle to this API is still pending. This functionality is delicate enough security-wise (especially with respect to different addresses on different forks) that I think it makes sense not to expose it directly, but rather have it be an operation executed via the init actor (who can then derive a fresh address via looking up the address of its caller). @zixuanzh @jbenet may also want to weigh in here

Reply:
Thanks, makes sense.
 I'll wait for @zixuanzh and @jbenet to pitch in before resolving it.